pkg/aws/awscontrolplane/admit_awscontrolplane_test.go

```diff
@@ -90,13 +103,28 @@ func TestAWSControlPlaneAdmit(t *testing.T) {
 			g8sControlplaneReplicas: nil,
 			expectAvailabilityZones: nil,
 			validAvailabilityZones:  []string{"eu-central-1a", "eu-central-1b", "eu-central-1c"},
+			currentInstanceType:     "m4.xlarge",
+			expectedInstanceType:    "",
+		},
+		{
+			// Here we check if the instance type is defaulted
+			name: "case 6",
+			ctx:  context.Background(),
+
+			currentAvailabilityZone: []string{"eu-central-1a", "eu-central-1b", "eu-central-1c"},
+			g8sControlplaneReplicas: nil,
+			expectAvailabilityZones: nil,
+			validAvailabilityZones:  []string{"eu-central-1a", "eu-central-1b", "eu-central-1c"},
+			currentInstanceType:     "",
+			expectedInstanceType:    aws.DefaultMasterInstanceType,
```

Comment:
Please note that this is still missing unit tests for the pre HA versions. I will add these later. (Here or in a separate PR - no preferences here)